infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -388,6 +391,37 @@ private LogData getLogData(LogEntry entry) {
         return set;
     }
 
+    private boolean isOverwritten(SegmentHandle fh, long filePosition) {
+        return fh.overwrittenFilePositions.contains(filePosition);
+    }
+
+
+    private void readOverwrittenPositions(SegmentHandle fh) throws IOException {
+        synchronized (fh.lock) {
+            File f = new File(fh.getOverwriteFileName());
+            if (f.exists()) {
+                try (DataInputStream is = new DataInputStream(new BufferedInputStream(new FileInputStream(fh.getOverwriteFileName())))) {
+                    while (is.available() > 4) {
```

Comment:
The segment lock is used because we can't read partial files (i.e. can't read while entries are partially written, we need a delimiter), so if a write is happening a lock would prevent reading and vice versa. Since you're code doesn't need that requirement I don't see why you need to use a lock here.

Also note, that there is only a single writer thread that is calling append, so you also dont need a lock in addOverwrittenPosition.